blockchain/My-Wallet-V3-iOS
# feat(KYC): IOS-1115 - Address entry screens

## Description

**Note: Marked DNM until some minor cleanup, comments, and keyboard fixes are added.**
This PR includes two new screens for users to input their address. The first screen, `KYCAddressController` leverages `MKLocalSearchCompleter` to provide address recommendations (e.g. autocomplete) to the user as they input their address.

* `LocationSuggestionService`: Conforms to `LocationSuggestionAPI`. It's the only class that knows about `MapKit`. As the user types the API returns a `LocationSearchResult`. This includes a `SearchUIState` as well as an array of `LocationSuggestion` objects. 
* Once the user taps a `LocationSuggestion` we fetch the full address from `MapKit` using `MKLocalSearchRequest`. This is an async request. Once the request completes, the user is routed to the `KYCAddressDetailViewController`.
* `KYCAddressDetailViewController` takes a `PostalAddress` model. It's from this model that we generate an array of `CellModel` types. There's only two at the moment, `PlainCell` and `TextEntryCell`. 
* Note that this PR also bumps our required iOS version to iOS 10.

** TODO: Handle passing along this data through the KYC flow. **

## How to Test

1. Build and run
2. Proceed through the KYC flow
3. When at the address screen, begin typing in an address.
4. Observe the address recommendations
5. Select an address
6. Observe the address detail screen

## Screenshots

<img width="333" alt="screen shot 2018-07-31 at 10 10 49 am" src="https://user-images.githubusercontent.com/41585563/43473281-55982e6e-94b5-11e8-9bd0-2de6e8a51b2f.png">
<img width="339" alt="screen shot 2018-07-31 at 10 49 08 am" src="https://user-images.githubusercontent.com/41585563/43473287-5835285c-94b5-11e8-9558-36025e337dd9.png">

## Additional Information

Ticket: `IOS-1115`
